torch/csrc/jit/passes/onnx/constant_map.h

```diff
@@ -66,6 +73,12 @@ class ConstantValueMap {
   // This map indicates whether the current type is estimated through inference
   // or tracer.
   std::unordered_map<std::string, bool> useInferredTypeMap;
+  // This map indicates a tensor value which represents a shape.
+  // The difference between shapeMap and shapeValueMap:
+  // shapeMap stores the shape of the tensor from a node.
+  // shapeValueMap stores the value of the tensor from a node when this tensor
```

Comment:
`when this tensor represents a shape` How can this be determined?

Reply:
The first tensor represents a shape is from a `Shape` node. Then we have a list of supported operators that actually propagate this shapeValueMap. And it will stop propagating when it reaches the `shape` of `Reshape` node.

Reply:
Is it possible that some of the nodes produces incompatible tensor with rank > 1?

Reply:
No. (1) Not every node supported shapeValueMap. From the node definition, we should know whether it supports or not. (2) When the node produces a tensor with rank > 1, then it should not be a shape value. So when we write the logic for the node, we should make sure that only the tensor with rank <=1 be considered as shape value.

Reply:
synced offline, added more comments to illustrate this:
  // We assume that the rank of the tensor value <= 1, and we ensure this when
  // we write the processing logic for the operators.